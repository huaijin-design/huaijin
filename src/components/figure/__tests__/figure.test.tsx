import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Figure } from '..';

describe('Figure', () => {
  it('renders correctly', () => {
    const json = renderer.create(
      <Figure figcaption="Depression. By: Darren Harris">
        <span>
          Depression is running through my head,<br/>
          These thoughts make me think of death,<br/>
          A darkness which blanks my mind,<br/>
          A walk through the graveyard, what can I find?....
        </span>
      </Figure>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });
});
