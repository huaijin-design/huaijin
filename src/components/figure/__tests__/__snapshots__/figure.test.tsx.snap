// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure renders correctly 1`] = `
.emotion-2 {
  position: relative;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin-top: 0.4em;
  padding: 0 1em;
  font-size: 0.9em;
  line-height: 2.2;
  text-align: center;
  color: #c7c7c7;
  position: relative;
}

.emotion-1::after {
  content: '';
  width: 20%;
  height: 1px;
  position: absolute;
  left: 40%;
  bottom: 0;
  background-color: rgba(255,111,97,0.2);
}

<figure
  className="emotion-2"
>
  <div
    className="emotion-0"
  >
    <span>
      Depression is running through my head,
      <br />
      These thoughts make me think of death,
      <br />
      A darkness which blanks my mind,
      <br />
      A walk through the graveyard, what can I find?....
    </span>
  </div>
  <figcaption
    className="emotion-1"
  >
    Depression. By: Darren Harris
  </figcaption>
</figure>
`;

exports[`Figure renders style rules -> effect 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  position: relative;
}

.emotion-2:hover > figcaption {
  background-color: rgba(64,64,64,0.8);
  opacity: 1;
  -webkit-transform: scale3d(1,1,1);
  -ms-transform: scale3d(1,1,1);
  transform: scale3d(1,1,1);
}

.emotion-1 {
  min-height: 40px;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  padding: 0.4em 1em 0.4em 0.4em;
  color: #fff;
  text-transform: uppercase;
  opacity: 0;
  -webkit-transition: opacity 0.35s,-webkit-transform 0.35s;
  -webkit-transition: opacity 0.35s,transform 0.35s;
  transition: opacity 0.35s,transform 0.35s;
  -webkit-transform: scale3d(1.4,1.4,1);
  -ms-transform: scale3d(1.4,1.4,1);
  transform: scale3d(1.4,1.4,1);
}

<figure
  className="emotion-2"
>
  <div
    className="emotion-0"
  >
    <span>
      Depression is running through my head,
      <br />
      These thoughts make me think of death,
      <br />
      A darkness which blanks my mind,
      <br />
      A walk through the graveyard, what can I find?....
    </span>
  </div>
  <figcaption
    className="emotion-1"
  >
    Depression. By: Darren Harris
  </figcaption>
</figure>
`;
